sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/HttpPipelineAdapter.java

```diff
@@ -57,4 +81,44 @@ public IHttpResponse send(HttpRequest httpRequest) throws Exception {
                 })))
             .block();
     }
+
+    private void logAccounIdentifiersIfConfigured(String body) {
+        if (!identityClientOptions.getIdentityLogOptionsImpl().isLoggingAccountIdentifiersAllowed()) {
+            return;
+        }
+        try {
+            JsonNode node = OBJECT_MAPPER.readTree(body);
+            JsonNode accessToken = node.get(ACCESS_TOKEN_JSON_KEY);
+            if (accessToken != null) {
+                String[] base64Metadata = accessToken.asText().split("\\.");
+                if (base64Metadata.length > 1) {
+                    byte[] decoded = Base64.getDecoder().decode(base64Metadata[1]);
+                    String data = new String(decoded, StandardCharsets.UTF_8);
+                    JsonNode dataNode = OBJECT_MAPPER.readTree(data);
+                    JsonNode appId = dataNode.get(APPLICATON_ID_JSON_KEY);
+                    JsonNode objectId = dataNode.get(OBJECT_ID_JSON_KEY);
+                    JsonNode tenantId = dataNode.get(TENANT_ID_JSON_KEY);
+                    JsonNode userPrincipalName = dataNode.get(USER_PRINCIPAL_NAME_JSON_KEY);
+
+                    CLIENT_LOGGER.log(LogLevel.INFORMATIONAL, () -> MessageFormat
+                        .format(ACCOUNT_IDENTIFIER_LOG_MESSAGE,
+                            getAccountIdentifierMessage(APPLICATION_IDENTIFIER, appId),
+                            getAccountIdentifierMessage(OBJECT_ID, objectId),
+                            getAccountIdentifierMessage(TENANT_ID, tenantId),
+                            getAccountIdentifierMessage(USER_PRINCIPAL_NAME,
+                                userPrincipalName)));
+                }
+            }
+        } catch (JsonProcessingException e) {
+            CLIENT_LOGGER.log(LogLevel.WARNING, () -> "allowLoggingAccountIdentifiers Log option was set,"
```

Comment:
Pass the exception down to the logger and let the logger handle exception logging instead of concatenating e.getMessage()

Reply:
done